src/peft/tuners/lora/config.py

```diff
@@ -69,6 +69,57 @@ class LoftQConfig:
     loftq_iter: int = field(default=1, metadata={"help": "Alternating iterations for LoftQ"})
 
 
+@dataclass
+class EvaConfig:
+    """
+    This is the sub-configuration class to store the configuration for a data-driven initialization via EVA. EVA was
+    introduced in <a href='https://arxiv.org/abs/2410.07170'>Explained Variance Adaptation</a>.
+
+    Args:
+        rho (`float`):
+            Rho value for EVA redistribution (>= 1.0). The maximum rank for a layer is lora_r * rho. Default is 2.0
+            meaning the maximum rank allowed for a layer is 2r. Increasing rho will allow for a higher degree of
+            redistribution of ranks across layers. Some pre-trained models might be more sensitive to a rank
+            redistribution. It can therefore be beneficial to try rho=1.0 if the performance is lower than expected.
+        tau (`float`):
+            Cosine similarity threshold for early stopping. Compares the cosine similarity of right-singular vectors
+            between two consecutive SVD steps. If the cosine similarity is above this threshold, the SVD iteration is
+            stopped. Default is 0.99.
+        use_label_mask (`bool`):
+            Use label mask for EVA initialization. This means that positions where labels=label_mask_value are ignored
+            for the SVD computation. Setting use_label_mask=True is preferred in most cases and can be especially
+            beneficial for multi-turn conversations. The default value is True. Filtering out items based on the label
+            mask can sometimes lead to a small batch size and as a result instabilities in the SVD computation. For
+            cases where a large share of batch items would be filtered out, set use_label_mask=False.
+        label_mask_value (`int`):
+            If use_label_mask=True the value to look for to mask out ignored tokens. Default is -100.
+        whiten (`bool`): Apply whitening to singular vectors. Default is False.
+            Whitening has been shown to be beneficial for EVA in the vision domain.
+        use_alpha_pattern (`bool`):
```

Comment:
done. We also did some checks and it seems in the language modeling setting, adjusting alpha seems to work a bit better. Since I would assume a large share of users use peft for language model tasks I set the default now to `True`